app/src/main/java/ch/epfl/sweng/swenggolf/database/Database.java

```diff
@@ -82,6 +82,29 @@ public abstract void write(@NonNull String path, @NonNull String id, @NonNull Ob
     public abstract <T> void read(@NonNull String path, @NonNull String id,
                                   @NonNull ValueListener<T> listener, @NonNull Class<T> c);
 
+    /**
+     * Perform changes every time an update is done in the database.
+     * It returns the new value using a listener.
+     *
+     * @param path     where to read the value.
+     * @param id       the name of the value field.
+     * @param listener the onDataChange method will be called at every change of the value if it
+     *                 exists and onCancelled will be called otherwise.
+     * @param c        the class of the value
+     * @param <T>      the type of the value
+     */
+    public abstract <T> void listen(@NonNull String path, @NonNull String id,
+                                    @NonNull ValueListener<T> listener, @NonNull Class<T> c);
+
+    /**
+     * Removes a listener from a reference.
+     *
+     * @param listener the listener to be removed.
+     * @param <T>      the type of the value.
+     */
+    public abstract <T> void deafen(@NonNull String path, @NonNull String id,
```

Comment:
I like that choice of name :100:

Reply:
Is it ironic ?

Reply:
No, I really like it! I might be tired but it made me laugh

Reply:
Okay, great if it stole you a smile !